hpx/runtime/parcelset/parcelport_impl.hpp

```diff
@@ -811,45 +791,19 @@ namespace hpx { namespace parcelset
             sender_connection->verify(parcel_locality_id);
 #endif
             // encode the parcels
-            std::size_t num_parcels = encode_parcels(&parcels[0],
-                parcels.size(), sender_connection->buffer_,
+            encode_parcels(&p,
+                std::size_t(-1), sender_connection->buffer_,
```

Comment:
To make a distinction between "always pass one parcel" and "we happen to just have one parcel, but we serialize the parcel count anyway". The latter is not used anymore right now because the batch serialization of parcels led to deadlocks with `invoke_when_ready` removed.